src/main.rs

```diff
@@ -25,15 +25,27 @@ enum NorgToken {
     Eof,
 }
 
-fn lexer() -> impl Parser<char, Vec<NorgToken>, Error = chumsky::error::Simple<char>> {
-    let special_chars = "*-~/_!%^,\"'$:@|=.#+>";
+const SPECIAL_CHARS: &str = "*-~/_!%^,\"'$:@|=.#+<>()[]{}";
+
+/// take a char and turn it into a NorgToken. Takes away the guessing game of 'is that char in the
+/// special chars list?'
+fn norg_char(char: char) -> NorgToken {
+    match char {
+        ' ' => NorgToken::Whitespace,
+        '\n' => NorgToken::SingleNewline,
+        _ if SPECIAL_CHARS.contains(char) => NorgToken::Special(char),
+        _ => NorgToken::Text(char.to_string()),
+    }
+}
```

Comment:
I may be wrong, but I don't see the motivation for this function :p

I see that this is only used in places where you want to match a bracket, in which case replace `norg_char(')')` with `Special(')')`. Doing so this way only adds a level of indirection. See the other review message for the `.map()` case.

Reply:
I made it for the map function mainly, and b/c originally `(` and `[` and a few others weren't special chars so I always had to guess if I should use `Text()` or `Special()`.

I agree that we can get rid of it if I just list out all the cases in a big select macro.